java-frontend/src/test/files/se/ObjectOutputStreamCheck.java

```diff
@@ -0,0 +1,52 @@
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.io.ObjectOutputStream;
+import java.io.OutputStream;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.nio.file.StandardOpenOption;
+import java.nio.file.OpenOption;
+
+class A {
+  void noncompliant_1(String fileName) throws IOException {
+    FileOutputStream fos = new FileOutputStream(fileName , true);  // fos opened in append mode
+    ObjectOutputStream out = new ObjectOutputStream(fos);  // Noncompliant
+  }
+  void noncompliant_2(String fileName, boolean appendMode) throws IOException {
+    if (!appendMode) return;
+    FileOutputStream fos = new FileOutputStream(fileName, appendMode);  // fos opened in append mode
+    ObjectOutputStream out = new ObjectOutputStream(fos);  // Noncompliant
+  }
+  void noncompliant_3(File file) throws IOException {
```

Comment:
You are not importing `java.io.File`, so I'm a bit surprise we are able to resolve this constructor.